src/contract/ens/eth_ens.rs

```diff
@@ -0,0 +1,409 @@
+
+        self.registry.ttl(node).await
+    }
+
+    /// Sets the caching TTL (time-to-live) of a name.
+    pub async fn set_ttl(&self, from: Address, domain: &str, ttl: u64) -> Result<TransactionId, ContractError> {
+        let domain = self.normalize_name(domain)?;
+        let node = namehash(&domain);
+
+        self.registry.set_ttl(from, node, ttl).await
+    }
+
+    /// Creates a new subdomain of the given node, assigning ownership of it to the specified owner.
+    pub async fn set_subnode_owner(
+        &self,
+        from: Address,
+        domain: &str,
+        label: &str,
+        owner: Address,
+    ) -> Result<TransactionId, ContractError> {
+        let domain = self.normalize_name(domain)?;
+        let node = namehash(&domain);
+
+        let label = self.normalize_name(label)?;
+        let label = crate::signing::keccak256(label.as_bytes());
+
+        self.registry.set_subnode_owner(from, node, label, owner).await
+    }
+
+    /// Sets the owner, resolver, and TTL for an ENS record in a single operation.
+    pub async fn set_record(
+        &self,
+        from: Address,
+        domain: &str,
+        owner: Address,
+        resolver: Address,
+        ttl: u64,
+    ) -> Result<TransactionId, ContractError> {
+        let domain = self.normalize_name(domain)?;
+        let node = namehash(&domain);
+
+        self.registry.set_record(from, node, owner, resolver, ttl).await
+    }
+
+    /// Sets the owner, resolver and TTL for a subdomain, creating it if necessary.
+    pub async fn set_subnode_record(
+        &self,
+        from: Address,
+        domain: &str,
+        label: &str,
+        owner: Address,
+        resolver: Address,
+        ttl: u64,
+    ) -> Result<TransactionId, ContractError> {
+        let domain = self.normalize_name(domain)?;
+        let node = namehash(&domain);
+
+        let label = self.normalize_name(label)?;
+        let label = crate::signing::keccak256(label.as_bytes());
+
+        self.registry
+            .set_subnode_record(from, node, label, owner, resolver, ttl)
+            .await
+    }
+
+    /// Sets or clears an approval.
+    /// Approved accounts can execute all ENS registry operations on behalf of the caller.
+    pub async fn set_approval_for_all(
+        &self,
+        from: Address,
+        operator: Address,
+        approved: bool,
+    ) -> Result<TransactionId, ContractError> {
+        self.registry.set_approval_for_all(from, operator, approved).await
+    }
+
+    /// Returns true if the operator is approved to make ENS registry operations on behalf of the owner.
+    pub async fn is_approved_for_all(&self, owner: Address, operator: Address) -> Result<bool, ContractError> {
+        self.registry.check_approval(owner, operator).await
+    }
+
+    /// Returns true if node exists in this ENS registry.
+    /// This will return false for records that are in the legacy ENS registry but have not yet been migrated to the new one.
```

Comment:
```suggestion
    /// Returns true if node exists in this ENS registry.
    ///
    /// This will return false for records that are in the legacy ENS registry but have not yet been migrated to the new one.
```